### ISSUE TITLE
api: implement Cloudflare Worker KV as a data store

### ISSUE THREAD

**Issue by iameli** [livepeer/livepeer-monorepo#453]
Alongside `LevelStore` and `PostgresStore` we should have a `CloudflareStore` that implements the same interface.

**Comment by angyangie** [comment]
**Notes from Discord:** 
This unblocks a lot, like intelligent API broadcaster routing, conditional per-user routing based on datacenter location, etc.

Currently, data on European streams would have to be stored in our Iowa Postgres database, so that's a big ugly round-trip hit every time. Instead, we'd like to move the database onto CloudFlare Worker KV, so we can look up the address of the pertinent broadcaster from anywhere! Pretty nice.


**Reasons for doing this:** 
- Lower latency when hitting /broadcaster on our API (or any other routes)?
- Eliminates the single point of failure that's the current Postgres server in Iowa, decreases latency for all regions that aren't Iowa, and unblocks us compiling the entire API server to a Cloudflare Worker which would dramatically increase availability and latency
- Currently, we can't do anything stateful in our existing worker (livepeerjs/packages/api/src/worker.js) because access to our database requires a connection to Postgres which you can't do in pure JavaScript on a worker. So incoming requests for manifests currently hit every single broadcaster to check to see who has the data.
- What I'd like to do instead is add logic to the auth webhook handler (POST /api/stream/hook) that remembers which broadcasters have handled a given manifestID. Then when we get a https://livepeer.live/stream/abc123.m3u8 request, we can ask for the manifests from the exact broadcasters that might have data for us instead of checking against every single broadcaster in the world. But we can't query that data from the worker unless we move from Postgres to Cloudflare Worker KV.

**Info on infrastructure:**
- The livepeerci/api Docker container runs in all of our Kubernetes clusters; that's the thing we've worked with locally. They're accessible at https://esh.livepeer.live/api, https://mcw-staging.livepeer-stagig.live/api and whatnot. Additionally, we compile a Cloudflare Worker from src/worker.js and deploy that to CloudFlare, which is accessible at https://livepeer.live and https://livepeer-staging.live.
Jumping into a different meeting, will continue explanation after
- The worker does a few things for us, mostly related to geolocation.

1. For requests to https://livepeer.live/api/broadcaster (or /orchestrator) it pings both Kubernetes clusters and races their responses, returning whichever one gets back first. So in the US, you'll probably get a bunch of mcw.livepeer.live Iowa broadcasters. You can see your local ping time at https://livepeer.live/api/geolocate, which is handled entirely by the worker and shows your ping time to the various regions, neat!

2. For requests to https://livepeer.live/stream/*, which is to say requests for media segments or manifests, it amalgamates from both clusters. So this way coherent manifests can be served from the worker that account for data from multiple regions.

3. (where is the webhook handler running?) For all other requests, such as the webhook handler, the worker just redirects the request to the nearest cluster. The broadcasters are all configured to hit the worker at https://livepeer.live/api/stream/hook but those requests get passed through to the API nodes running on the clusters themselves. Those clusters use the same (Iowa) Postgres database as their source of truth, so they have the same data generally.

**In summary**: we always hit CloudFlare worker livepeer.live/..., which then redirects request to either nearest API docker container node running on K8, or amalgamates response from /stream on all clusters. So, the CF worker acts as a "proxy". It can also handle requests entirely on its own (like https://livepeer.live/api/geolocate).

**Comment by wohlner** [comment]
Blocks https://app.zenhub.com/workspaces/engineering-5b60e4761ae1aa2935bcbdbc/issues/livepeer/livepeerjs/454 api: implement compilation to ServiceWorker/Cloudflare Worker
